crates/net/eth-wire-types/src/message.rs

```diff
@@ -299,6 +315,7 @@ impl<N: NetworkPrimitives> Encodable for EthMessage<N> {
             Self::NodeData(data) => data.encode(out),
             Self::GetReceipts(request) => request.encode(out),
             Self::Receipts(receipts) => receipts.encode(out),
+            Self::Other(unknown) => unknown.encode(out),
```

Comment:
I believe this isn't quite right because this now encodes the entire message but what we need to encode here is the message payload which is already encoded here so we only need to extend out with the unknown.payload

the way messages are encoded is like:

https://github.com/paradigmxyz/reth/blob/main/crates/net/eth-wire-types/src/message.rs#L122-L128